lib/Onnxifi/GlowOnnxManager.h

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_ONNXIFI_GLOWONNXMANAGER_H
+#define GLOW_ONNXIFI_GLOWONNXMANAGER_H
+
+#include <unordered_set>
+
+namespace glow {
+namespace onnxifi {
+class BackendId;
+class Backend;
+class Event;
+class Graph;
+
+/// Singleton class for creating and destroying objects for the ONNX interface.
+/// GlowOnnxManager tracks objects it has created and can be used to check if
+/// if an ONNX interface object was created by glow or not.
+class GlowOnnxManager {
+public:
+  /// Get a reference to the GlowOnnxManager singleton. There should only ever
+  /// be one GlowOnnxManager.
+  static GlowOnnxManager &get();
+
+  // Disallow copying GlowOnnxManager to help enforce singleton pattern.
+  GlowOnnxManager(const GlowOnnxManager &) = delete;
+  GlowOnnxManager &operator=(const GlowOnnxManager &) = delete;
```

Comment:
what about move copy ctor?
also, if we do not intend to inherit from the GlowOnnxManager we should make that `final` class.

Reply:
@rdzhabarov declaring the copy constructor as deleted counts as declaring it I believe so the condition "there are no user-declared copy constructors" is false. This example seems to confirm this https://ideone.com/Clhvy5

Reply:
I'm focusing on the wrong details here, you're right that it's better to have move assignment and move construction explicitly deleted, I marked them deleted.